src/cartesian/Bar.js

```diff
@@ -184,6 +186,58 @@ class Bar extends Component {
     return labelItem;
   }
 
+  renderErrorbars() {
+    const { isAnimationActive } = this.props;
+    if (isAnimationActive && !this.state.isAnimationFinished) { return null; }
+
+    const { data, errorBarFill, errorBarStrokeWidth } = this.props;
+    const errorKey = this.props.dataKey + 'Error';
+    const errorBars = data.map((entry, i) => {
+      let xMid, yMid, xMin, yMin, xMax, yMax, scale = 0;
+      let coordsTop, coordsMid, coordsBot = {};
+      let errorVal = entry[errorKey];
+
+      if (errorVal) {
+        if (this.props.layout === 'vertical') {
+          scale = entry.width / entry.value; 
```

Comment:
The scale may have `domain` like [10, 20]. And the scale of `XAxis` here can be very different, not only `linear`.